File: frontend/src/types/Lecture.ts
export const LECTURE_TYPE = {
	assignment: "assignment",
	forum: "forum",
	quiz: "quiz",
	resource: "resource",
	url: "url",
	page: "page",
	unknown: "unknown",
} as const;

export type LectureType = keyof typeof LECTURE_TYPE;

export type Lecture = {
	name: string;
	url: string;
	type: LectureType;
};

File: discord-bot/models/lecture.go
package models

type LectureType string

const (
	LectureResource   LectureType = "resource"
	LectureAssignment LectureType = "assignment"
	LectureUrl        LectureType = "url"
	LectureQuiz       LectureType = "quiz"
	LectureForum      LectureType = "forum"
	LecturePage       LectureType = "page"
	LectureUnknown    LectureType = "unknown"
)

type Lecture struct {
	Name     string      `json:"name"`
	Url      string      `json:"url"`
	Type     LectureType `json:"type"`
	Deadline int64       `json:"deadline"`
}
